From 5e7ec8e897ecff052c0524bb4704ef485142b99d Mon Sep 17 00:00:00 2001
From: Benjamin Valentin <benjamin.valentin@ml-pa.com>
Date: Wed, 1 Feb 2023 14:41:09 +0100
Subject: [PATCH] [PATCH 15/27] driver/i2c: remove the include gpio.h in i2c.h
 header

Including driver/i2c.h by RIOT code leads to type conflicts with RIOT gpio_t type.
---
 components/driver/i2c.c                | 1 +
 components/driver/include/driver/i2c.h | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/driver/i2c.c b/components/driver/i2c.c
index 164b9e9e6a..e703a6f6de 100644
--- a/components/driver/i2c.c
+++ b/components/driver/i2c.c
@@ -26,6 +26,7 @@
 #include "esp_rom_gpio.h"
 #include "esp_rom_sys.h"
 #include <sys/param.h>
+#include "driver/gpio.h"
 
 static const char *I2C_TAG = "i2c";
 
diff --git a/components/driver/include/driver/i2c.h b/components/driver/include/driver/i2c.h
index 57d9091e70..c55e5f4fcb 100644
--- a/components/driver/include/driver/i2c.h
+++ b/components/driver/include/driver/i2c.h
@@ -19,7 +19,6 @@ extern "C" {
 #include "freertos/task.h"
 #include "freertos/queue.h"
 #include "freertos/ringbuf.h"
-#include "driver/gpio.h"
 #include "soc/soc_caps.h"
 #include "hal/i2c_types.h"
 
-- 
2.37.2

